src/status.rs

```diff
@@ -458,6 +464,52 @@ impl StatusBuilder {
         self
     }
 
+    pub fn spacefed(mut self, spacefed: Spacefed) -> Self {
+        self.spacefed = Some(spacefed);
+        self
+    }
+
+    pub fn add_event(mut self, event: Event) -> Self {
+        if let Some(ref mut events) = self.events {
+            events.push(event);
+        } else {
+            self.events = Some(vec![event]);
+        }
```

Comment:
It works if `radio_show` implements `Copy`:
https://play.rust-lang.org/?gist=4c28f663874b4d16f7e36302fd60cc48&version=stable&backtrace=0
If not we need a clone:
https://play.rust-lang.org/?gist=89d89487c6a798d2eb5907556857c153&version=stable&backtrace=0

> The whole thing is surprisingly tricky for such a supposedly simple operation :D

Exactly what I was thinking: It can't be *that* hard, but apparently it is :wink: